datafusion/src/optimizer/eliminate_filter.rs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Optimizer rule to replace `where false` on a plan with an empty relation.
+//! This saves time in planning and executing the query.
+//! Note that this rule should be applied after simplify expressions optimizer rule.
+use datafusion_common::ScalarValue;
+use datafusion_expr::Expr;
+
+use crate::error::Result;
+use crate::logical_plan::plan::Filter;
+use crate::logical_plan::{EmptyRelation, LogicalPlan};
+use crate::optimizer::optimizer::OptimizerRule;
+
+use super::utils;
+use crate::execution::context::ExecutionProps;
+
+/// Optimization rule that elimanate the scalar value filter with an [LogicalPlan::EmptyRelation]
+#[derive(Default)]
+pub struct EliminateFilter;
+
+impl EliminateFilter {
+    #[allow(missing_docs)]
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl OptimizerRule for EliminateFilter {
+    fn optimize(
+        &self,
+        plan: &LogicalPlan,
+        execution_props: &ExecutionProps,
+    ) -> Result<LogicalPlan> {
+        match plan {
+            LogicalPlan::Filter(Filter { predicate, input }) => match predicate {
+                Expr::Literal(ScalarValue::Boolean(Some(v))) => {
+                    if *v {
+                        return Ok(LogicalPlan::Filter(Filter {
```

Comment:
Here we could also remove the filter and return the `input` logical plan?

Reply:
Yes, we can remove the filter which is `true`. But I'm not familiar with the datafusion optimizer, I don't know the good way for remove the filter

Reply:
I think we should be able to return `Ok(input.clone())`?